bouncer/shared/send_btc.ts

```diff
@@ -12,25 +12,33 @@ export const btcClient = new Client({
 });
 
 export async function selectInputs(amount: number) {
-  // List unspent UTXOs
-  const utxos = await btcClient.listUnspent();
+  return btcClientMutex.runExclusive(async () => {
+    // List unspent UTXOs
+    const utxos = await btcClient.listUnspent();
```

Comment:
We don't need to break up the vault swap construction into so many steps (in btc_vault_swap). Putting a mutex around just the input selection doesn't solve our problem. 

We should create method on the btc client that takes the nulldata utxo, deposit address and amount as args, and then does all the steps with the mutex acquired.

```
fn build_vault_tx(nulldata, amount, address):
	acquire_mutex()
		get_inputs()
		create_raw_tx()
		sign_tx()
		broadcast_tx()
	release_mutex()
```

In fact maybe only the first two steps need to be in the mutex, once we build the tx I guess the btc node is smart enough to know we can use the output (??) but to be safe we migth as well put everything inside.

Reply:
>  Putting a mutex around just the input selection doesn't solve our problem.

The mutex and the locking of the utxo should solve it. I don't understand.

> We should create method on the btc client

Ok, that's fine, but I don't see the need. Just so we don't have to lock the utxo?

Reply:
It's what Kyle said: if we lock the utxos, then request the payload, then build and send the tx, there is a chance that some other process fails because there are none available. We can eliminate this race condition quite easily though: We don't need to lock the inputs at all, do we? We can request the payload, then build and send the tx with whatever utxos are available at that point. 

I don't see why we wouldn't just do this once and do it properly instead of kicking the can down the road until the next time all the PRs are blocked.